include/AudioDevice.h

```diff
@@ -90,26 +90,28 @@ class AudioDevice
 
 	virtual void stopProcessing();
 
-protected:
-	// subclasses can re-implement this for being used in conjunction with
-	// processNextBuffer()
-	virtual void writeBuffer(const SampleFrame* /* _buf*/, const fpp_t /*_frames*/) {}
-
-	// called by according driver for fetching new sound-data
-	fpp_t getNextBuffer(SampleFrame* _ab);
 
 	// convert a given audio-buffer to a buffer in signed 16-bit samples
 	// returns num of bytes in outbuf
-	int convertToS16(const SampleFrame* _ab,
+	static int convertToS16(const SampleFrame* _ab,
 						const fpp_t _frames,
 						int_sample_t * _output_buffer,
+						ch_cnt_t channels,
```

Comment:
I believe @michaelgregorius simplified out this channel count in #7156

Reply:
Yes, `SampleFrame` is explicitly stereo because tons of code has already assumed this before my changes in https://github.com/LMMS/lmms/pull/7156 and multi-channel (it was called surround sound in the old code) was not really supported anyway. So this interface might be a bit contradictory if the channel count is intended to describe the number of mono channels in the buffer.

On a side note (and not really related to this PR): Does this conversion really belong in the `AudioDevice` class? The functionality is not really related solely to audio devices and other code might also be interested in such conversions. It could be moved out with regards to separation of concerns.

Reply:
So I should I replace `channels` with "2"?

Reply:
Why have a parameter if it must always be 2 anyway? You can just remove the parameter and assume stereo channels in the method's implementation.